mp/src/game/client/momentum/ui/HUD/hud_speedometer.cpp

```diff
@@ -26,93 +31,159 @@ using namespace vgui;
-    Color m_SecondaryColor;
+    Color m_LastJumpVelColor;
+
+    Label *m_pUnitsLabel, *m_pAbsSpeedoLabel, *m_pHorizSpeedoLabel, *m_pLastJumpVelLabel, 
+        *m_pStageEnterExitLabel, *m_pStageEnterExitComparisonLabel;
+
+    int m_defaultUnitsLabelHeight, m_defaultAbsSpeedoLabelHeight, m_defaultHorizSpeedoLabelHeight, 
+        m_defaultLastJumpVelLabelHeight, m_defaultStageEnterExitLabelHeight;
+
+    int m_defaultYPos;
+
+    void DeactivateLabel(Label *label);
+    void ActivateLabel(Label *label, int defaultHeight);
 
-    bool m_bRanFadeOutJumpSpeed;
     CMomRunStats *m_pRunStats;
     CMomRunEntityData *m_pRunEntData;
 
+    ConVarRef m_cvarTimeScale;
   protected:
     CPanelAnimationVar(Color, m_bgColor, "BgColor", "Blank");
     // NOTE: These need to be floats because of animations (thanks volvo)
     CPanelAnimationVar(float, m_fStageStartAlpha, "StageAlpha", "0.0"); // Used for fading
-    CPanelAnimationVar(float, m_fLastJumpAlpha, "JumpAlpha", "0.0");
+    CPanelAnimationVar(float, m_fLastJumpVelAlpha, "JumpAlpha", "0.0");
 };
 
-DECLARE_NAMED_HUDELEMENT(CHudSpeedMeter, HudSpeedMeter);
+DECLARE_HUDELEMENT(CHudSpeedMeter);
 
 CHudSpeedMeter::CHudSpeedMeter(const char *pElementName)
-    : CHudElement(pElementName), CHudNumericDisplay(g_pClientMode->GetViewport(), "HudSpeedMeter")
+    : CHudElement(pElementName), EditablePanel(g_pClientMode->GetViewport(), "HudSpeedMeter"), 
+    m_cvarTimeScale("mom_replay_timescale")
 {
     ListenForGameEvent("zone_exit");
     ListenForGameEvent("zone_enter");
     SetProportional(true);
     SetKeyBoardInputEnabled(false);
     SetMouseInputEnabled(false);
     SetHiddenBits(HIDEHUD_LEADERBOARDS);
+    m_pRunStats = nullptr;
+    m_pRunEntData = nullptr;
     m_bRanFadeOutJumpSpeed = false;
+    m_flNextColorizeCheck = 0;
+    m_fStageStartAlpha = 0.0f;
+
+    m_flLastVelocity = 0;
+    m_flLastHVelocity = 0;
+    m_flLastJumpVelocity = 0;
+
+    m_iRoundedVel = 0;
+    m_iRoundedHVel = 0;
+    m_iRoundedLastJumpVel = 0;
+
+    m_pUnitsLabel = new Label(this, "UnitsLabel", "");
+    m_pAbsSpeedoLabel = new Label(this, "AbsSpeedoLabel", "");
+    m_pHorizSpeedoLabel = new Label(this, "HorizSpeedoLabel", "");
+    m_pLastJumpVelLabel = new Label(this, "LastJumpVelLabel", "");
+    m_pStageEnterExitLabel = new Label(this, "StageEnterExitLabel", "");
+    m_pStageEnterExitComparisonLabel = new Label(this, "StageEnterExitComparisonLabel", "");
+    
+    LoadControlSettings("resource/ui/Speedometer.res");
+
+    m_defaultAbsSpeedoLabelHeight = m_pAbsSpeedoLabel->GetTall();
+    m_defaultHorizSpeedoLabelHeight = m_pHorizSpeedoLabel->GetTall();
+    m_defaultLastJumpVelLabelHeight = m_pLastJumpVelLabel->GetTall();
+    m_defaultUnitsLabelHeight = m_pUnitsLabel->GetTall();
+    m_defaultStageEnterExitLabelHeight = m_pStageEnterExitLabel->GetTall();
+
+    m_pStageEnterExitComparisonLabel->SetFont(m_pStageEnterExitLabel->GetFont()); //need to have same font
+
+    m_defaultYPos = GetYPos();
```

Comment:
Duplicate code found in `ApplySchemeSettings`, can this be removed?

Reply:
Or if you really need to initialize them, pull this and the code in `ApplySchemeSettings` out to a private member function.

Reply:
Ah, didn't realize `ApplySchemeSettings` was called after constructor. I was under the impression it was called when using `hud_reloadscheme`.
Removed this useless code.